### ISSUE TITLE
Ship's continue to do any action if action key is released in pause menu

### ISSUE THREAD

**Issue by Max5377** [pioneerspacesim/pioneer#5584]
### Observed behaviour
Ship continue to do any action if holding any action key(movement, fire, etc.), entering pause menu, releasing key there and resuming the game. If pressing this key again during game after unpausing, the ship will stop doing it.

https://github.com/pioneerspacesim/pioneer/assets/69468517/c28ebded-1ade-4385-8ab0-c47957d9faec

### Expected behaviour
Ship should stop do any action even if key is released in pause menu.

### Steps to reproduce
1. Hold any action key;
2. Open pause menu;
3. Release action key;
4. Resume game.

__My pioneer version (and OS):__
https://github.com/pioneerspacesim/pioneer/commit/e77da966923a4ad24c8863bcca7dec7933c33d43 (Windows 10 x64)

**Comment by impaktor** [comment]
For the record, you might want to play current master, that has bug fixes since february release, scroll down here (when it's done in a few minutes): https://github.com/pioneerspacesim/pioneer/actions/runs/5012626778

**Comment by Web-eWorks** [comment]
This is (arguably) intentional, and mostly intended to allow keyboard flight users to interact with the console screen / menus which require text inputs while still flying their ship. I have this on my list to re-evaluate along with another pass over the input subsystem.

**Comment by impaktor** [comment]
could it be an optional setting?

**Comment by Web-eWorks** [comment]
Not really, no. This is an extremely low-level behavior (do we pass key release events to input bindings that are shadowed/disabled on the input stack) and there's little to no benefit from making it toggleable. I intend to address this issue soon, as it's involved in other IMGUI-related input issues.

**Comment by Max5377** [comment]
To add to that, if I hold right mouse key (which gets cursor to hide) to rotate my ship and enter pause menu, the cursor will stay hidden there.

**Comment by Max5377** [comment]
In this function `void GuiApplication::PollEvents()` in `GuiApplication.cpp` since `PiGui` doesn't handle input the same way as `m_input`, it will think that key is still pressed after exiting the pause menu, since, as a quick fix, commenting `case SDL_KEYUP:` in `if (PiGui::WantCaptureKeyboard()) {` stops behaviour described in this issue.
